### PR TITLE
User can like another post

### PR DESCRIPTION
**Pull request by rithreaksa**
[User can like another post](https://www.pivotaltracker.com/story/show/176624562)
### User Story
```
As an  API,            
In order to send like count to the client,         
I want to store the like count.
```
## Changes proposed in this pull request:
* Adds `Like` controller and model
* Adds `Like` Factorybot
* Adds association between `Like` and `User` and `Post`
* Implements create action for like
* Writes RSpec test for like feature

## What I have learned working on this feature:
* Implements `Like` count feature with create action
* To test RSpec with `Like` feature

### CODE REVIEW

File: app/controllers/api/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Api::LikesController < ApplicationController
+  def create
+    if already_liked?
+      render json: { message: "You can't like more than once" }, status: 422
```
Comment (ViktorHek) [inline app/controllers/api/likes_controller.rb:4]: we are not supposed to use this message in the front end right...??

File: config/routes.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 Rails.application.routes.draw do
-  mount_devise_token_auth_for 'User', 
-  at: 'auth', 
-  controllers: { omniauth_callbacks: 'api/omniauth_callbacks' }
+  mount_devise_token_auth_for 'User', at: 'auth', controllers: { omniauth_callbacks: 'api/omniauth_callbacks' }
```
Comment (ViktorHek) [inline config/routes.rb]: I would prefer to move this back to the way it was with "at:" and "controllers:" on different lines. just for readability

File: spec/requests/api/user_can_like_another_post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    it 'is expected to return a 201 status' do
+      expect(response).to have_http_status 201
+    end
+
+    it 'is expected to increase like count to 1' do
+      expect(existing_post.likes.count).to eq 1
+    end
+  end
+  describe 'unsuccessfully when user already liked the post' do
```
Comment (ViktorHek) [inline spec/requests/api/user_can_like_another_post_spec.rb]: add a line here or remove extra lines between it blocks

Comment (rithreaksa) [reply spec/requests/api/user_can_like_another_post_spec.rb]: Okay 👍

File: config/routes.rb (hunk 2 above)
Comment (rithreaksa) [reply config/routes.rb]: it is the formatter.

File: app/controllers/api/likes_controller.rb (hunk 1 above)
Comment (rithreaksa) [reply app/controllers/api/likes_controller.rb:4]: It is not a necessity to display it in the frontend!

File: config/routes.rb (hunk 2 above)
Comment (KyunginNa) [reply config/routes.rb]: We put them in different lines though👍

**Pull request merged**